Compass/LoginViewController.swift

```diff
@@ -0,0 +1,216 @@
+        navigationItem.title = "No storyboard"
+        self.view.backgroundColor = UIColor.darkGray
+        
+        titleLabel.textColor = UIColor.red
+        
+        loginButton.addTarget(self, action: #selector(loginButtonPressed), for: .touchUpInside)
+        
+        switchPageButton.setTitle("Not registered? SignUp", for: .normal)
+        switchPageButton.addTarget(self, action: #selector(SwitchPagePressed), for: .touchUpInside)
+        
+        NametextField.placeholder = "e-mail address..."
+        NametextField.text = "test@mail.co.uk"
+        NametextField.textColor = UIColor.white
+        NametextField.layer.backgroundColor = UIColor.blue.cgColor
+        NametextField.attributedPlaceholder = NSAttributedString(string: "Name...", attributes: [NSAttributedStringKey.foregroundColor: UIColor.white])
+        
+        
+        passTextField.placeholder = "password..."
+        passTextField.text = "pass"
+        passTextField.textColor = UIColor.white
+        passTextField.isSecureTextEntry = true
+        passTextField.layer.backgroundColor = UIColor.blue.cgColor
+        passTextField.attributedPlaceholder = NSAttributedString(string: "Password...", attributes: [NSAttributedStringKey.foregroundColor : UIColor.white])
+        
+        
+        let stackView = UIStackView(arrangedSubviews: [titleLabel, NametextField, passTextField, loginButton, switchPageButton])
+        stackView.axis = .vertical
+        stackView.spacing = 10;
+        
+        if isloginPage {
+            titleLabel.text = "Login Page"
+            loginButton.setTitle("Login", for: .normal)
+            switchPageButton.setTitle("Go to Signup page", for: .normal)
+            
+        } else {
+            titleLabel.text = "Signup Page"
+            loginButton.setTitle("Signup", for: .normal)
+            switchPageButton.setTitle("Go to Login page", for: .normal)
+        }
+        
+        self.view.addSubview(stackView)
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([stackView.widthAnchor.constraint(equalToConstant: 200),
+                                     stackView.centerXAnchor.constraint(lessThanOrEqualTo: self.view.centerXAnchor),
+                                     stackView.centerYAnchor.constraint(lessThanOrEqualTo: self.view.centerYAnchor)])
+        
+        
+        
+    }
+    
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+    }
+    
+    @objc private func loginButtonPressed() {
+        
+        if NametextField.text == "" || passTextField.text == "" {
+            createAlert(title: "Invalid username or password format", message: "Please enter valid username or password")
+        } else if !isValidEmail(testStr: NametextField.text!) {
+            createAlert(title: "Invalid e-mail address", message: "Please enter valid e-mail")
+        } else {
```

Comment:
Rather than doing if then else for these checks, we would typically make seperate ifs and make them return e.g.:
```
if NametextField.text == "" || passTextField.text == "" {
            createAlert(title: "Invalid username or password format", message: "Please enter valid username or password")
            return
}

if !isValidEmail(testStr: NametextField.text!) {
            createAlert(title: "Invalid e-mail address", message: "Please enter valid e-mail")
            return
}
```

Or better yet, you could use a guard. Since you don't want any of the rest of the function to happen if these checks happen, we would typically use a guard here just to make the meaning explicit (we try to always use guards for this kind of check) E.g.:
```
guard NametextField.text != "", passTextField.text != "" else {
     createAlert(title: "Invalid username or password format", message: "Please enter valid username or password")
     return
}
```

Or since you need the values later, you can use guard let again:
```
guard let name = NametextField.text, name.count > 0, 
          password = passTextField.text, password.count > 0 else {
     createAlert(title: "Invalid username or password format", message: "Please enter valid username or password")
     return
}
```
This last one is admittedly more complicated, but is probably the version I would write